src/auth/authSlice.js

```diff
@@ -0,0 +1,32 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const authSlice = createSlice({
+  name: 'auth',
+  initialState: {
+    authToken: localStorage.getItem('FITNESS_TRACKER_APP_ADMIN_TOKEN') || null,
+    username: localStorage.getItem('username') || '',
+    error: null,
+  },
+  reducers: {
+    setAuthToken: (state, action) => {
+      state.authToken = action.payload;
+    },
+    setUsername: (state, action) => {
+      state.username = action.payload;
+    },
+    setError: (state, action) => {
+      state.error = action.payload;
+    },
+    clearError: (state) => {
```

Comment:
remove this, you can handle this in extra reducers since the request is in **thunks**.

Pls check how to handle **thunks** in the slice from here (https://redux-toolkit.js.org/api/createAsyncThunk) or check how it was implemented in the **wbemt project**